app/main/data_requests.py

```diff
@@ -23,6 +23,7 @@ def post_ingest(file: FileStorage, data: dict = None) -> tuple[bool, dict | None
     files = {"excel_file": (file.name, file, MIMETYPE.XLSX)}
 
     response = requests.post(request_url, files=files, data=data)
+    file.seek(0)  # reset the stream position
```

Comment:
Sorry to flip-flop on this but maybe it's more weird to have this out of context here? Does the same happen if we open the file in an `open` block?

Reply:
An `open` block is not applicable here because the file is already stored in memory - it's never saved to disk.

As far as I'm aware, if you read a file stream then you have to reset the stream position to the beginning before reading it again - otherwise when you come to read it again it will be read as an empty file.

Interested to see a better solution, if there is one.